File: LeetCode/0070-climbing-stairs/0070-climbing-stairs.js
/**
 * @param {number} n
 * @return {number}
 */
var climbStairs = function(n) {
    var DP = Array.from({length: n+1}, (_, i) => 0);
    DP[1] = 1;
    DP[2] = 2;
    for (let i = 3; i < n+1; i++) {
        DP[i] = DP[i-2] + DP[i-1];
    }
    return DP[n];
};

File: LeetCode/0070-climbing-stairs/0070-climbing-stairs.py
class Solution:
    def climbStairs(self, n: int) -> int: 
        DP = [0 for _ in range(n)]
        DP[0] = 1
        if n >= 2: DP[1] = 2
        for i in range(2, n):
            DP[i] = DP[i-1]+DP[i-2]
        return DP[n-1]